### ISSUE TITLE
Error for ref in condition

### ISSUE THREAD

**Issue by maicss** [vuejs/eslint-plugin-vue#2140]
**Please describe what the rule should do:**

Should give an error when user use a ref value as condition assert.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
const someValue = ref(false)
// this condition always true, should give an error, suggest add `.value` after the variable
if (someValue) {

}
// should give an error too
[1, someValue].includes()
[1, someValue].some()
[1, someValue].every()
```

**Additional context**

No additional context

**Comment by maicss** [comment]
It seems we should not use ref without unref anywhere...like arguments for functions.

**Comment by jhoermann** [comment]
You should use [@typescript-eslint/no-unnecessary-condition](https://typescript-eslint.io/rules/no-unnecessary-condition/).

**Comment by FloEdelmann** [comment]
A rule catching this is indeed hard to get right without TypeScript, so I'd also suggest [@typescript-eslint/no-unnecessary-condition](https://typescript-eslint.io/rules/no-unnecessary-condition/). Duplicating (part of) this rule in eslint-plugin-vue seems to be much work with little gain, while always missing some cases that should be reported.

**Comment by maicss** [comment]
Thanks! I will try this out.
